File: app/src/main/java/com/gago/david/myland/Models/TaskObject.java
package com.gago.david.myland.Models;

import java.io.Serializable;
import java.util.Date;
import java.util.Objects;

public class TaskObject implements Serializable {
    public long rowid;
    public String land;
    public Integer plantIndex;
    public String taskType;
    public Integer priority;
    public Date creationDate;
    public Date targetDate;
    public boolean completed;
    public String observations;

    public TaskObject(String land, Integer plantIndex, String taskType, Integer priority, Date creationDate, Date targetDate, boolean completed, String observations) {
        this.land = land;
        this.plantIndex = plantIndex;
        this.taskType = taskType;
        this.priority = priority;
        this.creationDate = creationDate;
        this.targetDate = targetDate;
        this.completed = completed;
        this.observations = observations;
    }

    public TaskObject(long rowid, String land, Integer plantIndex, String taskType, Integer priority, Date creationDate, Date targetDate, boolean completed, String observations) {
        this.rowid = rowid;
        this.land = land;
        this.plantIndex = plantIndex;
        this.taskType = taskType;
        this.priority = priority;
        this.creationDate = creationDate;
        this.targetDate = targetDate;
        this.completed = completed;
        this.observations = observations;
    }

    @Override
    public String toString() {
        return "TaskObject{" +
                "rowid=" + rowid +
                ", land='" + land + '\'' +
                ", plantIndex=" + plantIndex +
                ", taskType='" + taskType + '\'' +
                ", priority=" + priority +
                ", creationDate=" + creationDate +
                ", targetDate=" + targetDate +
                ", completed=" + completed +
                ", observations='" + observations + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        TaskObject that = (TaskObject) o;
        return rowid == that.rowid &&
                completed == that.completed &&
                Objects.equals(land, that.land) &&
                Objects.equals(plantIndex, that.plantIndex) &&
                Objects.equals(taskType, that.taskType) &&
                Objects.equals(priority, that.priority) &&
                Objects.equals(creationDate, that.creationDate) &&
                Objects.equals(targetDate, that.targetDate) &&
                Objects.equals(observations, that.observations);
    }

    @Override
    public int hashCode() {

        return Objects.hash(rowid, land, plantIndex, taskType, priority, creationDate, targetDate, completed, observations);
    }

    public TaskObject clone() {
        return new TaskObject(rowid, land, plantIndex, taskType, priority, creationDate, targetDate, completed, observations);
    }
}

File: app/src/main/java/com/gago/david/myland/models/TaskObject.kt
package com.gago.david.myland.models

import java.io.Serializable
import java.util.*

class TaskObject : Serializable, Cloneable {
    var rowid: Long = 0
    var land: String
    var plantIndex: Int?
    var taskType: String
    var priority: Int
    var creationDate: Date
    var targetDate: Date?
    var completed: Boolean
    var observations: String
    var completedDate: Date? = null

    constructor(land: String, plantIndex: Int?, taskType: String, priority: Int, creationDate: Date, targetDate: Date?, completed: Boolean, observations: String) {
        this.land = land
        this.plantIndex = plantIndex
        this.taskType = taskType
        this.priority = priority
        this.creationDate = creationDate
        this.targetDate = targetDate
        this.completed = completed
        this.observations = observations
    }

    constructor(
        rowid: Long,
        land: String,
        plantIndex: Int?,
        taskType: String,
        priority: Int,
        creationDate: Date,
        targetDate: Date?,
        completed: Boolean,
        observations: String
    ) {
        this.rowid = rowid
        this.land = land
        this.plantIndex = plantIndex
        this.taskType = taskType
        this.priority = priority
        this.creationDate = creationDate
        this.targetDate = targetDate
        this.completed = completed
        this.observations = observations
    }

    constructor(
        rowid: Long,
        land: String,
        plantIndex: Int?,
        taskType: String,
        priority: Int,
        creationDate: Date,
        targetDate: Date?,
        completed: Boolean,
        observations: String,
        completedDate: Date?
    ) {
        this.rowid = rowid
        this.land = land
        this.plantIndex = plantIndex
        this.taskType = taskType
        this.priority = priority
        this.creationDate = creationDate
        this.targetDate = targetDate
        this.completed = completed
        this.observations = observations
        this.completedDate = completedDate
    }

    override fun toString(): String {
        return "TaskObject{" +
                "rowid=" + rowid +
                ", land='" + land + '\'' +
                ", plantIndex=" + plantIndex +
                ", taskType='" + taskType + '\'' +
                ", priority=" + priority +
                ", creationDate=" + creationDate +
                ", targetDate=" + targetDate +
                ", completed=" + completed +
                ", observations='" + observations + '\'' +
                ", completedDate='" + completedDate + '\'' +
                '}'
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as TaskObject
        return rowid == that.rowid && completed == that.completed &&
                land == that.land &&
                plantIndex == that.plantIndex &&
                taskType == that.taskType &&
                priority == that.priority &&
                creationDate == that.creationDate &&
                targetDate == that.targetDate &&
                observations == that.observations &&
                completedDate == that.completedDate
    }

    override fun hashCode(): Int {
        return Objects.hash(rowid, land, plantIndex, taskType, priority, creationDate, targetDate, completed, observations, completedDate)
    }

    public override fun clone(): TaskObject {
        return TaskObject(rowid, land, plantIndex, taskType, priority, creationDate, targetDate, completed, observations, completedDate)
    }
}